src/bindings/bnd_mesh.cpp

```diff
@@ -717,24 +717,33 @@ BND_TUPLE BND_MeshFaceList::GetFace(int i) const
 
 BND_TUPLE BND_MeshFaceList::GetFaceVertices(int faceIndex) const
 {
-  ON_MeshFace& face = m_mesh->m_F[faceIndex];
-  BND_TUPLE rc = CreateTuple(5);
-  SetTuple<bool>(rc, 0, true);
-  for (int i = 0; i < 4; i++)
-    SetTuple<ON_3fPoint>(rc, i+1, m_mesh->m_V[ face.vi[i] ]);
-  return rc;
+  int count = m_mesh->m_F.Count();
+  if(faceIndex >= 0 && faceIndex < count)
+  {
+    ON_MeshFace& face = m_mesh->m_F[faceIndex];
+    BND_TUPLE rc = CreateTuple(5);
+    SetTuple<bool>(rc, 0, true);
+    for (int i = 0; i < 4; i++)
+      SetTuple<ON_3fPoint>(rc, i+1, m_mesh->m_V[ face.vi[i] ]);
+    return rc;
+  }
+  return NullTuple();
 }
 
 ON_3dPoint BND_MeshFaceList::GetFaceCenter(int faceIndex) const
 {
-  ON_3dPoint center;
-  ON_MeshFace& face = m_mesh->m_F[faceIndex];
-  if (face.IsQuad())
-    center = 0.25 * (m_mesh->m_V[face.vi[0]] + m_mesh->m_V[face.vi[1]] + m_mesh->m_V[face.vi[2]] + m_mesh->m_V[face.vi[3]]);
-  else
-    center = (1.0 / 3.0) * (m_mesh->m_V[face.vi[0]] + m_mesh->m_V[face.vi[1]] + m_mesh->m_V[face.vi[2]]);
+  int count = m_mesh->m_F.Count();
+  if(faceIndex >= 0 && faceIndex < count)
+  {
+    ON_3dPoint center;
+    ON_MeshFace& face = m_mesh->m_F[faceIndex];
+    if (face.IsQuad())
+      center = 0.25 * (m_mesh->m_V[face.vi[0]] + m_mesh->m_V[face.vi[1]] + m_mesh->m_V[face.vi[2]] + m_mesh->m_V[face.vi[3]]);
+    else
+      center = (1.0 / 3.0) * (m_mesh->m_V[face.vi[0]] + m_mesh->m_V[face.vi[1]] + m_mesh->m_V[face.vi[2]]);
 
-  return center;
+    return center;
+  }
```

Comment:
It looks like you need to return a value in the case where faceIndex is invalid.  You could return ON_3dPoint::Unset for this case